test/metabase/driver/sql/query_processor_test_util.clj

```diff
@@ -0,0 +1,207 @@
+  []
+  (alter-var-root #'hformat/format-clause (constantly orig-format-clause)))
+
+(defmulti ^:private pretty-format
+  "Special version of [[hformat/format-clause]] that indents and pretty-prints the resulting form."
+  {:arglists '([clause sql-map])}
+  (.dispatchFn orig-format-clause))
+
+(defn install-pretty-formatters!
+  "Install code so HoneySQL will pretty-print the SQL it generates."
+  []
+  (alter-var-root #'hformat/format-clause (constantly pretty-format)))
+
+(defn- pretty-formatters-installed?
+  "Whether pretty formatters were installed by [[install-pretty-formatters!]] or not."
+  []
+  (identical? hformat/format-clause pretty-format))
+
+(defn do-with-pretty-formatters
+  "Impl for [[with-pretty-formatters]]."
+  [thunk]
+  (if (pretty-formatters-installed?)
+    (thunk)
+    (try
+      (install-pretty-formatters!)
+      (thunk)
+      (finally
+        (remove-pretty-formatters!)))))
+
+(defmacro with-pretty-formatters
+  "Execute body with the pretty formatters from [[install-pretty-formatters!]] temporarily installed (if they are not
+  already installed)."
+  [& body]
+  `(do-with-pretty-formatters (fn [] ~@body)))
+
+(def ^:private ^:dynamic *indent* 0)
+
+(defn- indent [] (str/join (repeat (* *indent* 2) \space)))
+
+(defn- newline-and-indent []
+  (str \newline (indent)))
+
+(def ^:private ^:dynamic *indent-comma-join* true)
+
+(defn- indent-comma-join [strs]
+  (if *indent-comma-join*
+    (str (newline-and-indent)
+         (str/join (str "," (newline-and-indent))
+                   strs))
+    (str/join ", " strs)))
+
+(defmethod pretty-format :default
+  [clause sql-map]
+  (str
+   (newline-and-indent)
+   (binding [*indent*            (inc *indent*)
+             *indent-comma-join* true]
+     (let [to-sql hformat/to-sql]
+       (with-redefs [hformat/comma-join indent-comma-join
+                     hformat/to-sql     (fn [form]
+                                          (binding [*indent-comma-join* false]
+                                            (to-sql form)))]
+         (orig-format-clause clause sql-map))))))
+
+
+;;;; [[query->sql-map]] and [[sql->sql-map]] -- these parse an actual SQL map into a pseudo-HoneySQL form
+
+(defn pretty-sql
+  "Remove quotes around identifiers (where possible) and remove `public.` qualifiers."
+  [s]
+  (if-not (string? s)
+    s
+    (-> s
+        (str/replace #"\"([\w\d_-]+)\"" "$1")
+        (str/replace #"PUBLIC\." "")
+        (str/replace #"public\." ""))))
+
+(defn even-prettier-sql
+  "Do [[pretty-sql]] transformations, and remove excess whitespace and *all* quote marks."
+  [s]
+  (-> s
+      pretty-sql
+      (str/replace #"`" "")
+      (str/replace #"\s+" " ")
+      (str/replace #"\(\s*" "(")
+      (str/replace #"\s*\)" ")")
+      (str/replace #"'" "\"")
+      str/trim))
+
+(defn- symbols [s]
+  (binding [*read-eval* false]
+    (read-string (str \( s \)))))
+
+(def ^:private sql-keywords
+  '#{[LEFT JOIN]
```

Comment:
I understand this is a best effort, but an actual complete list will kinda make you want to rip your own face off

https://en.wikipedia.org/wiki/SQL_reserved_words

Reply:
It's really just the ones we want to put on new lines, and only for tests at that, so it doesn't have to be a huge list. I should probably mention that in a comment

Reply:
I renamed this to `sql-keywords-that-should-get-newlines` and added a comment as well to hopefully make the purpose a little clearer here.